test/plugins/controls/connectivity/pauseOnError/test.js

```diff
@@ -0,0 +1,173 @@
+        );
+    });
+
+    QUnit.cases
+        .init([
+            { title: 'init' },
+            { title: 'render' },
+            { title: 'finish' },
+            { title: 'destroy' },
+            { title: 'trigger' },
+            { title: 'getTestRunner' },
+            { title: 'getAreaBroker' },
+            { title: 'getConfig' },
+            { title: 'setConfig' },
+            { title: 'getState' },
+            { title: 'setState' },
+            { title: 'show' },
+            { title: 'hide' },
+            { title: 'enable' },
+            { title: 'disable' }
+        ])
+        .test('plugin API ', (data, assert) => {
+            const runner = getRunner();
+            const plugin = pluginFactory(runner);
+            assert.equal(
+                typeof plugin[data.title],
+                'function',
+                `The pluginFactory instance exposes a "${data.title}" function`
+            );
+        });
+
+    QUnit.test('error then reload', assert => {
+        const done = assert.async();
+        const runner = getRunner();
+        Promise.resolve()
+            .then(() => new Promise((resolve, reject) => {
+                const timeout = window.setTimeout(() => {
+                    reject(new Error('The plugin did not catch the error or did not complete properly!'));
+                }, 200);
+                runner
+                    .on('init', () => {
+                        assert.ok(true, 'Runner has been initialized');
+                        runner.trigger('error');
+                    })
+                    .on('confirm.*', (message, accept, cancel, options) => {
+                        assert.equal(typeof message, 'string', 'String message provided');
+                        assert.equal(typeof accept, 'function', 'Accept callback provided');
+                        assert.equal(typeof cancel, 'function', 'Cancel callback provided');
+                        assert.equal(typeof options, 'object', 'Options provided');
+                        cancel();
+                    })
+                    .before('pause', () => {
+                        assert.ok(false, 'The pause should not be triggered!');
+                        runner.destroy();
+                        return Promise.reject();
+                    })
+                    .before('reloadpage', () => {
+                        assert.ok(true, 'A page reload has been triggered');
+                        runner.destroy();
+                        return Promise.reject();
+                    })
+                    .on('destroy', () => {
+                        window.clearTimeout(timeout);
+                        resolve();
+                    })
+                    .init();
+            }))
+            .catch(err => {
+                assert.pushResult({
+                    result: false,
+                    message: err
+                });
+            })
+            .then(done);
```

Comment:
Sorry for the long rewrite, but I do not understand the purpose of using Promise. Isn't my solution is the same?

- Assertion count check could check that every assert was run. No setTimeout necessary.
- Why do you return with Promise.reject in event handler? `false` is enough, isn't it?

```suggestion
                assert.expect(6);
                runner
                    .on('init', () => {
                        assert.ok(true, 'Runner has been initialized');
                        runner.trigger('error');
                    })
                    .on('confirm.*', (message, accept, cancel, options) => {
                        assert.equal(typeof message, 'string', 'String message provided');
                        assert.equal(typeof accept, 'function', 'Accept callback provided');
                        assert.equal(typeof cancel, 'function', 'Cancel callback provided');
                        assert.equal(typeof options, 'object', 'Options provided');
                        cancel();
                    })
                    .before('pause', () => {
                        assert.ok(false, 'The pause should not be triggered!');
                        runner.destroy();
                    })
                    .before('reloadpage', () => {
                        assert.ok(true, 'A page reload has been triggered');
                        runner.destroy();
                        return false;
                    })
                    .on('destroy', done)
                    .init();
```

Reply:
>  Assertion count check could check that every assert was run. No setTimeout necessary.
Indeed, but in this case, the unit test will hang if the tested code does not perform as expected. It is better to control the flow.

>  Why do you return with Promise.reject in event handler? false is enough, isn't it?
`return false` is the legacy way. A compatibility layer still allows it, but the proper way is to use a `Promise.reject`.

Reply:
> Assertion count check could check that every assert was run. No setTimeout necessary.
> Indeed, but in this case, the unit test will hang if the tested code does not perform as expected. It is better to control the flow.

I do not agree. It makes the test more complicated for no reason. Most of the time the test runs successfully, so there is no hang at all. But if you would like to avoid long hanging, then set a shorter timeout for the test.

Reply:
Well, controlling the flow of a unit test allows making sure the expected behavior is performing properly, and there is no unknown. This is also especially useful in the context of TDD.
But anyway, I agree this construction is not useful for this particular case. At least removing it doesn't harm.